core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -160,15 +160,20 @@ private[spark] abstract class Task[T](
 
   // A flag to indicate whether the task is killed. This is used in case context is not yet
   // initialized when kill() is invoked.
-  @volatile @transient private var _killed = false
+  @volatile @transient private var _maybeKillReason: String = null
```

Comment:
Any reason to make this a String and not Option[String] - like other places it is defined/used ?